http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -78,9 +78,17 @@ class ClusterHelper(dbRef: DbReference,
             IO.unit
           } else {
             for {
-              emails <- getDataprocServiceAccounts(googleProject)
-              _ <- if (emails.isEmpty) IO.raiseError(ClusterIamSetupException(imageProject))
-                   else emails.traverse(e => retryIam(imageProject, e, Set("roles/compute.imageUser")))
+              projectNumber <- IO.fromFuture(IO(googleComputeDAO.getProjectNumber(googleProject))).map(_.getOrElse(throw ClusterIamSetupException(imageProject)))
+              roles = Set("roles/compute.imageUser")
+
+              // The Dataproc SA is used to retrieve the image. However projects created prior to 2016
```

Comment:
Last change to this code, I promise. Instead of adding _both_ service accounts, try to add the dataproc one first, and add the API one if that fails. Google confirmed that only projects created prior to 2016 do _not_ have a dataproc SA, which matches what I have seen.